.github/testflinger/job-def.yaml

```diff
@@ -0,0 +1,116 @@
+job_queue: REPLACE_QUEUE
+output_timeout: 7200
+provision_data:
+  REPLACE_PROVISION_DATA
+test_data:
+  test_cmds: |
+
+    # Exit immediately if a test fails
+    set -e
+
+    # Clone repo from appropriate branch/commit.
+    ssh ubuntu@$DEVICE_IP '
+      sudo DEBIAN_FRONTEND=noninteractive apt update
+      sudo DEBIAN_FRONTEND=noninteractive apt -y upgrade
+      sudo DEBIAN_FRONTEND=noninteractive apt -y install git curl
+      git clone -b REPLACE_BRANCH \
+        https://github.com/canonical/intel-npu-driver-snap.git \
+        ~ubuntu/intel-npu-driver-snap
+      cd ~ubuntu/intel-npu-driver-snap
+      echo "Current git branch: $(git branch --show-current)"
+      echo "Latest commit:"
+      git log --name-status HEAD^..HEAD
+    '
+
+    # Install dependencies
+    ssh ubuntu@$DEVICE_IP '
+      sudo snap install --classic snapcraft
+      sudo snap install lxd --channel=5.21/stable
+      sudo adduser ubuntu lxd
+      sudo snap refresh
+    '
+
+    # Disable swap and IPv6
+    ssh ubuntu@$DEVICE_IP '
+      sudo sysctl -w vm.swappiness=0
+      sudo echo "vm.swappiness = 0" | sudo tee -a /etc/sysctl.conf
+      sudo swapoff -a
+      echo "net.ipv6.conf.all.disable_ipv6=1" | sudo tee -a /etc/sysctl.conf
+      echo "net.ipv6.conf.default.disable_ipv6=1" | sudo tee -a /etc/sysctl.conf
+      echo "net.ipv6.conf.lo.disable_ipv6=1" | sudo tee -a /etc/sysctl.conf
+      sudo sysctl -p
+    '
+
+    # Build and install intel-npu-driver snap
+    ssh ubuntu@$DEVICE_IP '
+      lxd init --auto
+      cd ~ubuntu/intel-npu-driver-snap
+      snapcraft
+      sudo snap install --dangerous ./intel-npu-driver_*_amd64.snap
+      sudo snap connect intel-npu-driver:intel-npu-fw
```

Comment:
Do we still need to manually do these connections ? in the smoke-tests, i see that you expect the auto-connection. Does the auto-connection depends on `snapd` version ?

Reply:
When you install from a locally built snap the autoconnections do not come up so you need to connect them manually. For the smoke tests, there is some sort of magic in the `action-build` GitHub action that allows the autoconnections to come up. I asked about this in Mattermost but didn't receive a response.

Reply:
@frenchwr Got it, can you try to compare the snapd version in the runner and on the device ?

Reply:
Both are on 2.63